python/src/aiconfig/registry.py

```diff
@@ -10,21 +10,37 @@
 
 
 class ModelParserRegistry:
-    # A dictionary to store registered model parsers by their IDs
+    """
+    A dictionary to store registered model parsers by their IDs. It stores both:
+        1) model_name --> model_parser (ex: "mistralai/Mistral-7B-v0.1" --> 
+            HuggingFaceTextGenerationTransformer)
+        2) moder_parser.id() --> model_parser (ex: HuggingFaceTextGenerationTransformer 
+            works with many different Text Generation models) so is saved as 
+            HuggingFaceTextGenerationTransformer.id() (str) (usually it's classname) --> 
+            HuggingFaceTextGenerationTransformer (obj) 
+        
+    The reason we allow 2) above is so that you can define the relationship between a 
+    model name and a model parser in the AIConfig instead of the model parser class (ex:
+    https://github.com/lastmile-ai/aiconfig/blob/main/cookbooks/HuggingFace/Mistral-aiconfig.json#L16-L18).
+    This then gets updated into the desired format 1) by calling the 
+    `update_model_parser_registry_with_config_runtime()` command
+    
+    Each model_name is only allowed to map to a single model_parser for an AIConfigRuntime.
+    """
     _parsers: Dict[str, ModelParser] = {}
 
     @staticmethod
-    def register_model_parser(model_parser: ModelParser, ids: List[str] = None):
+    def register_model_parser(model_parser: ModelParser, model_names: List[str] = None):
```

Comment:
Wouldn't this be a breaking change on the AIConfig API? I think we should revert back to `ids`, not worth a breaking change

Reply:
Yea it would. I feel it's worth updating since it's more clear though. I can just update the docstrings for now though